util/clustering/kmeans_clustering.py

```diff
@@ -0,0 +1,231 @@
+# Based on the blog post "How to cluster images based on visual similarity" 
+# written by Gabe Flomo.
+
+
+# for loading/processing the images  
+from keras.preprocessing.image import load_img 
+from keras.preprocessing.image import img_to_array 
+from keras.applications.vgg16 import preprocess_input 
+
+# models 
+from keras.applications.vgg16 import VGG16 
+from keras.models import Model
+
+# clustering and dimension reduction
+from sklearn.cluster import KMeans
+from sklearn.decomposition import PCA
+
+# for everything else
+import argparse
+import os
+import numpy as np
+import matplotlib.pyplot as plt
+from PIL import Image
+from tqdm import tqdm
+
+
+def main():
+    parser = argparse.ArgumentParser(description='Clustering')
+    if 'SM_OUTPUT_DATA_DIR' in os.environ: 
+
+        parser.add_argument('--output-data-dir', type=str, default=os.environ['SM_OUTPUT_DATA_DIR'])
```

Comment:
'SM_OUTPUT_DATA_DIR' is completely managed by SageMaker backend, we don't want that (we want to control where our files are going). For now, set this argument to have no default (we can discuss this later when we put it on SageMaker).